lumicks/pylake/population/mixture.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+from sklearn.mixture import GaussianMixture
+from scipy import stats
+
+
+def as_sorted(fcn):
+    """Decorator to return results sorted according to mapping array."""
+
+    def wrapper(self, *args, **kwargs):
+        result = fcn(self, *args, **kwargs)
+        return result[self._map]
+
+    return wrapper
+
+
+class GaussianMixtureModel:
+    def __init__(self, data, K, init_method, n_init, tol, max_iter):
+        """A wrapper around scikit-learn's GMM
+
+        Parameters
+        ----------
+        K : int
+            number of Gaussian components in the model
+        """
+        self.K = K
+        self._model = GaussianMixture(
+            n_components=K, init_params=init_method, n_init=n_init, tol=1e-3, max_iter=100
+        )
+        self._model.fit(data)
```

Comment:
Would it make sense to keep a reference to the data it was fitted on? In that case, the trace argument of plot could be optional.